kernel/src/main/java/org/kframework/parser/KRead.java

```diff
@@ -51,4 +58,21 @@ public static K deserialize(String stringToParse, InputModes inputMode) {
                 throw KEMException.criticalError("Unsupported input mode: " + inputMode);
         }
     }
+
+    public static K autoDeserialize(byte[] kast, Source source) {
+        if (BinaryParser.isBinaryKast(kast))
+            return BinaryParser.parse(kast);
+
+        try {
```

Comment:
can we just check that the first non-whitespace character in the input is '{'? instead of trying to parse it, failing, and then trying to parse it again differently? it will be a lot faster that way.

Reply:
Sure, that makes more sense.